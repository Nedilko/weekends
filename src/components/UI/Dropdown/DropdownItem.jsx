import PropTypes from 'prop-types'
import { useEffect, useRef } from 'react'

DropdownItem.propTypes = {
  handleSelect: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  isSelected: PropTypes.bool.isRequired,
}

function DropdownItem({ handleSelect, value, isSelected }) {
  const ref = useRef(null)

  useEffect(() => {
    if (isSelected && ref.current) {
      ref.current.scrollIntoView()
    }
  }, [isSelected])

  return (
    <option
      ref={ref}
      onClick={(e) => handleSelect(e.target.value)}
      value={value}
      data-testid="dropdown-item"
      className={`px-2 py-1 text-sm hover:bg-slate-200 dark:hover:bg-slate-700 ${
        isSelected ? 'font-regular' : 'font-thin'
      }`}
    >
      {value}
    </option>
  )
}

export default DropdownItem
