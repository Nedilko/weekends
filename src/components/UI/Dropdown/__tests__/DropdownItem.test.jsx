import DropdownItem from '@UI/Dropdown/DropdownItem'
import { render, screen, vi, userEvent } from '@utils/test-utils'

describe('button', () => {
  beforeAll(() => {
    window.HTMLElement.prototype.scrollIntoView = vi.fn()
  })

  afterEach(() => {
    window.HTMLElement.prototype.scrollIntoView.mockClear()
  })

  it('should render', () => {
    const handleSelect = vi.fn()
    render(
      <DropdownItem
        value="sample"
        handleSelect={handleSelect}
        isSelected={true}
      />
    )
    const dropdownItem = screen.getByText('sample')
    expect(dropdownItem).toBeInTheDocument()
  })

  it('should render selected', () => {
    const handleSelect = vi.fn()
    render(
      <DropdownItem
        value="sample"
        handleSelect={handleSelect}
        isSelected={true}
      />
    )
    const dropdownItem = screen.getByText('sample')
    expect(dropdownItem).toMatchSnapshot()
  })

  it('should render not selected', () => {
    const handleSelect = vi.fn()
    render(
      <DropdownItem
        value="sample"
        handleSelect={handleSelect}
        isSelected={false}
      />
    )
    const dropdownItem = screen.getByText('sample')
    expect(dropdownItem).toMatchSnapshot()
  })

  it('should handle click', async () => {
    const handleSelect = vi.fn()
    render(
      <DropdownItem
        value="sample"
        handleSelect={handleSelect}
        isSelected={true}
      />
    )
    const user = userEvent.setup()
    const dropdownItem = screen.getByText('sample')
    await user.click(dropdownItem)
    expect(handleSelect).toHaveBeenCalledTimes(1)
  })
})
