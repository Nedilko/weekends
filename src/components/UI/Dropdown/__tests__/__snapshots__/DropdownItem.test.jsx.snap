// Vitest Snapshot v1

exports[`button > should render not selected 1`] = `
<option
  class="px-2 py-1 text-sm hover:bg-slate-200 dark:hover:bg-slate-700 font-thin"
  value="sample"
>
  sample
</option>
`;

exports[`button > should render selected 1`] = `
<option
  class="px-2 py-1 text-sm hover:bg-slate-200 dark:hover:bg-slate-700 font-regular"
  value="sample"
>
  sample
</option>
`;
